Makefile

```diff
@@ -244,88 +236,79 @@ modules: ## Runs go mod to ensure proper vendoring.
 .PHONY: generate
 generate: ## Generate code
 	$(MAKE) generate-go
-	$(MAKE) generate-manifests
 	$(MAKE) generate-test-flavors
 	$(MAKE) $(CRD_DOCS)
 
-.PHONY: $(CRD_DOCS_DIR)/%
-$(CRD_DOCS_DIR)/%:
+.PHONY:
+$(CRD_DOCS_DIR)/%: $(API_SRCS)
 	$(MAKE) -C docs/book src/crd/$*
 
 .PHONY: generate-go
 generate-go: $(MOCKGEN)
-	$(MAKE) generate-go-core
-	$(MAKE) generate-go-eks-bootstrap
-	$(MAKE) generate-go-eks-controlplane
+	$(MAKE) generate-go-apis
 	go generate ./...
 
-.PHONY: generate-go-core
-generate-go-core: ## Runs Go related generate targets
-	$(MAKE) -B $(CONTROLLER_GEN) $(DEFAULTER_GEN) $(CONVERSION_GEN)
+.PHONY: generate-go-apis
+generate-go-apis: ## Runs Go related generate targets
+	$(MAKE) .build/generate-go-apis
+
+.build:
+	mkdir -p .build
+
+.build/generate-go-apis: .build $(API_SRCS) $(CONTROLLER_GEN) $(DEFAULTER_GEN) $(CONVERSION_GEN)
 	$(CONTROLLER_GEN) \
 		paths=./api/... \
 		paths=./$(EXP_DIR)/api/... \
-		object:headerFile=./hack/boilerplate/boilerplate.generatego.txt
+		paths=./bootstrap/eks/api/... \
+		paths=./controlplane/eks/api/... \
+		paths=./iam/api/... \
+		output:crd:dir=config/crd/bases \
+		object:headerFile=./hack/boilerplate/boilerplate.generatego.txt \
+		crd:crdVersions=v1 \
+		rbac:roleName=manager-role \
+		webhook
 
 	$(CONTROLLER_GEN) \
-		paths=./cmd/clusterawsadm/api/... \
+		paths=./cmd/... \
 		object:headerFile=./hack/boilerplate/boilerplate.generatego.txt
 
-	$(DEFAULTER_GEN) \
-		--input-dirs=./cmd/clusterawsadm/api/bootstrap/v1alpha1,./cmd/clusterawsadm/api/iam/v1alpha1 \
-		--v=0 $(GEN_OUTPUT_BASE) \
-		--go-header-file=./hack/boilerplate/boilerplate.generatego.txt
+	$(MAKE) defaulters
 
 	$(CONVERSION_GEN) \
 		--input-dirs=./api/v1alpha3 \
+		--input-dirs=./api/v1alpha4 \
+		--input-dirs=./cmd/clusterawsadm/api/bootstrap/v1alpha1 \
 		--extra-peer-dirs=sigs.k8s.io/cluster-api/api/v1alpha3 \
-		--build-tag=ignore_autogenerated_core_v1alpha3 \
+		--extra-peer-dirs=sigs.k8s.io/cluster-api/api/v1alpha4 \
+		--build-tag=ignore_autogenerated_core \
 		--output-file-base=zz_generated.conversion $(GEN_OUTPUT_BASE) \
 		--go-header-file=./hack/boilerplate/boilerplate.generatego.txt
 
 	$(CONVERSION_GEN) \
 		--input-dirs=./$(EXP_DIR)/api/v1alpha3 \
-		--output-file-base=zz_generated.conversion \
-		--go-header-file=./hack/boilerplate/boilerplate.generatego.txt $(GEN_OUTPUT_BASE)
-
-.PHONY: generate-go-eks-bootstrap
-generate-go-eks-bootstrap: $(CONTROLLER_GEN)
-	$(CONTROLLER_GEN) \
-		paths=./bootstrap/eks/api/... \
-		object:headerFile=./hack/boilerplate/boilerplate.generatego.txt
-	$(CONVERSION_GEN) \
-		--input-dirs=./bootstrap/eks/api/v1alpha3 \
+		--input-dirs=./$(EXP_DIR)/api/v1alpha4 \
+		--extra-peer-dirs=./api/v1alpha3 \
```

Comment:
Tests are giving this error `conversion.go:310] failed to find pkg: ./api/v1alpha3`
I think instead of 
```
--extra-peer-dirs=./api/v1alpha3 \
--extra-peer-dirs=./api/v1alpha4 \
```
it should be 
```
--input-dirs=./api/v1alpha3 \
--input-dirs=./api/v1alpha4 \
```